### DISCUSSION TITLE
Ignore building static pages while I run `next build `

### DISCUSSION THREAD

**Discussion by taoqf** [vercel/next.js#36882] [Ideas]
### Describe the feature you'd like to request

I wish there will be a config to disable building static pages when I build the project.
But I with It could run first time on the server or when revalidate is timeout.
The reason is: I can not connect to the data server when I build the project.
And I could not build the pages on the server manually.

### Describe the solution you'd like

If there is an option to disabled this would be nice, maybe could be a build option with command `next build`

### Describe alternatives you've considered

Or could be a flag in .env

**Comment by balazsorban44** [comment]
Do you mean the `fallback: true` option of `getStaticPaths`?

This lets you skip static page generation and defer it until the first user hits a certain page, showing a fallback page for the first user, and then always show a static page (with optional [regeneration](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration)).

Please see the docs for more information:

https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-true

**Comment by taoqf** [comment]
I would rather not running `getStaticProps` at building time at all, not just `fallback`

**Comment by taoqf** [comment]
How do you guys build when you could not connect to database(in develop environment)?
pull the codes and run build script on server?

**Comment by danielcavanagh** [comment]
i'm having a similar issue. i was planning on pulling a large list of paths dynamically from a cms to prebuild and return the list from `getStaticPaths`. however a large list will cause `next build` to take forever, blocking deployment of the instance

the full build could be delayed til after deploy, but it's a bit odd to have nothing to serve at deploy time and it would lead to a brief period where client requests fail

`fallback: true` allows for a quick build and then a static build at request time, but it doesn't alone solve the core issue cause it doesn't allow for anything to be prebuilt

it would be great to be able to build the core files needed to serve requests during build, deploy the instance, then prebuild all static paths in the background

this would solve op's issue too i believe: on dev, build only core files; on prod, build static pages after core files

**Comment by danielcavanagh** [comment]
i'm thinking for now the best option is to use `fallback: true` with no list of paths, and run a separate post-deploy script that makes a web request for each path to force the build for each

**Reply by taoqf** [reply]
It's not enough. `fallback` is only used for dynamic routes.
I could not connect to db at all in my case.

**Reply by danielcavanagh** [reply]
yeah, fair enough. i was just talking about my situation

the only good solution would be a `next build --no-static` option to not build static files at all, with a `next build --static-only` for later to finish off the build when you know it'll work

**Reply by tago-SE** [reply]
I would also love this or a way to write code blocks that is able to distinguish between when it is building and when it is running normally.

**Comment by taoqf** [comment]
This is still not a way to use static pages in my app even in appdir because of this.

**Comment by zaru** [comment]
I am building an admin panel, so I do not need static page builds. I am using AppRouter, but sometimes it is automatically determined as static due to dynamic detection. To address this, I have added export `const dynamic = 'force-dynamic'` in my layout.tsx file.

However, since RouteHandlers do not allow for shared processing, I have to add `export const dynamic = 'force-dynamic'` for all GET requests, which is cumbersome.

It would be convenient if there was a setting to specify the build mode for the entire app.

**Reply by 9Dave9** [reply]
WOW.. this is awful, thanks for sharing, had no clue.

**Reply by JamieGoodson** [reply]
It looks like `export const dynamic = "force-dynamic";` isn't working in Next 15. Ffs I just want to deploy my app without build errors, I don't care about statically rendering this one particular page - for the love of god let me disable it! 😩
